sdk/mongocluster/arm-mongocluster/review/arm-mongocluster-api-mongoClusters.api.md

```diff
@@ -0,0 +1,91 @@
+## API Report File for "@azure/arm-mongocluster"
```

Comment:
Looks like it's the same compared to https://github.com/kazrael2119/azure-sdk-for-js/blob/networkanalytics-test/sdk/networkanalytics/arm-networkanalytics/review/arm-networkanalytics-api-dataProducts.api.md
But I also notices that when `hierarchyClient == true`, the codegen will do extra tasks like resolving operation groups.
If we don't want it, how about add a step to validate tsp config file in sdk autogen pipeline?

Reply:
Yeah, we can summary a few rules out,
1. we should disable hierarchyClient for every mgmt plane libraries but not for data plane, 
2. experiementalExtensibleEnums should be always true for mgmt plane non first preview libraries.
 just not sure if it's feasible? because I am not sure if pipeline task will parse the tspconfig.yaml file?

Reply:
I think it's feasible. After I enable modular for spec repo,  I can verify it for mgmt plane that similar to RLC: https://github.com/Azure/azure-sdk-tools/blob/614c99e8f442e7d64945a62a8cd800c48d9d9c74/tools/js-sdk-release-tools/src/llc/generateRLCInPipeline/generateRLCInPipeline.ts#L53

Reply:
Feel free to have a try, I have been asked to look into js generation failure because of service team miss package-dir configuration in their tspconfig.yaml for many times.  https://github.com/Azure/azure-rest-api-specs/pull/28981#discussion_r1615752653

Reply:
@qiaozha I may not get the point here because I notice the default value for hierarchyClient is true https://github.com/Azure/autorest.typescript/blob/main/packages/typespec-ts/src/transform/transfromRLCOptions.ts#L191-L200

Reply:
> you mean service team should explicitly set them as false for mgmt?

This is what I meant, hierarchyClient is default true for third party.

Reply:
> Feel free to have a try, I have been asked to look into js generation failure because of service team miss package-dir configuration in their tspconfig.yaml for many times. [Azure/azure-rest-api-specs#28981 (comment)](https://github.com/Azure/azure-rest-api-specs/pull/28981#discussion_r1615752653)

Will try.